src/backend/distributed/transaction/transaction_management.c

```diff
@@ -657,12 +657,22 @@ SwallowErrors(void (*func)())
 {
 	MemoryContext savedContext = CurrentMemoryContext;
```

Comment:
Should `savedContext` also be volatile?
```suggestion
	volatile MemoryContext savedContext = CurrentMemoryContext;
```